nuxeo-features/nuxeo-automation/nuxeo-automation-server/src/main/java/org/nuxeo/ecm/automation/server/jaxrs/adapters/AsyncOperationAdapter.java

```diff
@@ -171,12 +172,15 @@ public OperationException onError(OperationException error) {
 
         // TODO NXP-26303: use thread pool
         new Thread(() -> {
+            LoginComponent.pushPrincipal(principal);
             TransactionHelper.runInTransaction(() -> {
                 try (CloseableCoreSession session = CoreInstance.openCoreSession(repoName, principal)){
```

Comment:
Could you move the `LoginComponent.pushPrincipal(principal)` just before the `try`? That's the correct pattern, for instance if there was an exception in `runInTransaction` then the `finally` block wouldn't be executed.

Reply:
I made the requested change.